docs/scanners/semgrep.md

```diff
@@ -0,0 +1,22 @@
+# [semgrep](https://semgrep.dev)
+
+[`semgrep`](https://semgrep.dev) (syntactic grep) is an open-source tool for finding patterns in code. It's useful for preventing the use of known anti-patterns in a codebase or enforcing the correct use of secure-by-default frameworks (e.g. always use a project's sanitization method on user-provided data).
+
+semgrep is fast and powerful; it's grep-esque patterns are lifted into AST matchers. Compared to regexes these patterns aren’t affected by whitespaces, comments, newlines, the order of keyword arguments, variable renaming, and other language nuances.
+
+If a found pattern is forbidden this scanner will fail and the `message` will be show to the developer in the report. A `required` pattern must be found in order for the scan to pass
+
```

Comment:
Could you add the languages supported?  Also, add some info on config options like --exclude-dir, --exclude, --include, like in https://github.com/coinbase/salus/blob/master/docs/scanners/pattern_search.md?